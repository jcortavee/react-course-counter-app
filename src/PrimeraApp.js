import { Fragment } from "react";
import PropTypes from 'prop-types';

// Functional components
const PrimeraApp = ({greeting, subtitle}) => {
    // const greeting = 'Hola mundo cost';

    return (
        <Fragment>
            <h1>{ greeting }</h1>
            <p>{ subtitle }</p>
        </Fragment>
    );
};

PrimeraApp.propTypes = {
    greeting: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitle: 'Its a subtitle'
}

export default PrimeraApp;