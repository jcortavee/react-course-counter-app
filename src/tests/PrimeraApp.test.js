import React from 'react';
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Test in PrimeraApp', () => {

    // test('should show the message Hi, We are Paramore', () => {
    //    const greeting = 'Hi, We are Paramore';
    //    const wrapper = render(<PrimeraApp greeting={greeting} />);
    
    //    expect(wrapper.getByText(greeting)).toBeInTheDocument();
    // });
    
    test('should show PrimeraApp', () => {
        const greeting = 'Hi, We are Paramore';
        const wrapper = shallow(<PrimeraApp greeting={greeting} />);
        
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should show the subtitle sent in props', () => {
        const greeting = 'Hi, We are Paramore';
        const subtitle = 'Subtitle';
        const wrapper = shallow(<PrimeraApp greeting={greeting} subtitle={subtitle} />);
        
        const pText = wrapper.find('p').text();

        expect(pText).toBe(subtitle);
    })
    
    
    

});