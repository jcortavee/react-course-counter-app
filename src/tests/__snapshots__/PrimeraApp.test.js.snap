// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in PrimeraApp should show PrimeraApp 1`] = `
<Fragment>
  <h1>
    Hi, We are Paramore
  </h1>
  <p>
    Its a subtitle
  </p>
</Fragment>
`;
